carbon-design-system/carbon
# docs(storybook): avoid setting iframe html dir to empty string

In most cases when you click around the storybook deploy (and the storybook `Container` is unmounted), you will end up applying an empty `dir` attribute to the `html` tag inside the storybook iframe.

However, `dir` or `dir=""` aren't valid, and DAP will flag it with the following violation:

```
Valid values for the 'dir' attribute are 'ltr', 'rtl' or 'auto'.
Invalid value used for the 'dir' attribute.
```


![Screen Shot 2020-05-22 at 5 54 29 PM](https://user-images.githubusercontent.com/9057921/82714721-82de9580-9c55-11ea-82ff-11e1b7647b98.png)



⚠️  *Please note in the screenshot that you need to select the "Storybook" tab to see this violation.*


I've proposed a quick fix. Please let me know what you think. Even though `dir="auto"` is valid, I opted to *remove*  the `dir` attribute when there isn't an `originalDirection` to fall back to, because the [DAP help docs on the `dir` violation](https://aat.w3ibm.mybluemix.net/token/00000000-0000-0000-0000-000000000000/00000000-0000-0000-0000-000000000000/archives/2020JanDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1056.html) say this:

> HTML 5 introduced `auto` where the text direction is determined based on the content. Authors are urged to only use this value as a last resort when the direction of the text is truly unknown and no better server-side heuristic can be applied.

#### Changelog

**Changed**

- update logic of storybook container to avoid adding an empty `dir` attribute to the iframe's `html` tag

**Removed**

- there was an import & a variable that weren't being used in the file


#### Testing / Reviewing

1. Go to the storybook deploy: https://carbon-components-react.netlify.app/
2. **Click on *any* other component story in the left nav bar** -- this is important so that the `Container` is unmounted, and the code being changed here will run.
3. Run the DAP plugin using the latest Jan 2020 ruleset.
4. In the DAP plugin UI, make sure to select the "Storybook" tab, as show below:
![Screen Shot 2020-05-22 at 5 54 44 PM](https://user-images.githubusercontent.com/9057921/82714717-7d814b00-9c55-11ea-8986-5fb70df2544b.png)
